tidb-binlog/pump_client/pump.go

```diff
@@ -64,21 +98,40 @@ type PumpStatus struct {
 
 	grpcConn *grpc.ClientConn
 
+	reCreateClient bool
+
 	// the client of this pump
 	Client pb.PumpClient
 
-	ErrNum int64
+	state *realPumpState
 }
 
 // NewPumpStatus returns a new PumpStatus according to node's status.
 func NewPumpStatus(status *node.Status, security *tls.Config) *PumpStatus {
 	pumpStatus := PumpStatus{
-		Status:   *status,
-		security: security,
+		Status:         *status,
+		security:       security,
+		reCreateClient: true,
+		state:          &realPumpState{},
 	}
 	return &pumpStatus
 }
 
+// MarkReCreateClient marks that we need to re-create grpc connection
+func (p *PumpStatus) MarkReCreateClient() {
```

Comment:
why add a new variable reCreateClient, seems can execute ResetGrpcClient()

Reply:
???

Reply:
can execute ResetGrpcClient here, can judge whether should create client by judge "client == nil", why use `reCreateClient `